src/main/java/seedu/cakecollate/ui/CommandBox.java

```diff
@@ -69,6 +75,68 @@ private void setStyleToIndicateCommandFailure() {
         styleClass.add(ERROR_STYLE_CLASS);
     }
 
+    /**
+     * Adds the command typed in by the user to the array list userInputs.
+     * @param input the command that the user has typed in.
+     */
+    public void updateUserInputs(String input) {
+        userInputs.add(input);
+        incrementUserInputsIndex();
+        userInputsIndex = userInputs.size() - 1;
+        firstDecrementAfterUserInput = true;
+        positionCaretInTheEnd();
+    }
+
+    private void incrementUserInputsIndex() {
+        if (userInputsIndex + 1 < userInputs.size() && userInputsIndex < userInputs.size()) {
+            userInputsIndex++;
+        }
+    }
+
+    private void decrementUserInputsIndex() {
+        if (!firstDecrementAfterUserInput && userInputsIndex > 0) {
+            userInputsIndex--;
+        }
+        firstDecrementAfterUserInput = false;
+    }
+
+    public void setCommandTextField(Optional<String> input) {
+        input.ifPresent(string -> getCommandTextField().setText(string));
+        positionCaretInTheEnd();
+    }
+
+    private void positionCaretInTheEnd() {
+        getCommandTextField().positionCaret(getTextInCommandTextField().length());
+    }
+
+    public Optional<String> getPreviousInput() {
+        decrementUserInputsIndex();
+        Optional<String> previous = Optional.ofNullable(input());
+        return previous;
+    }
+
+    public Optional<String> getNextInput() {
+        incrementUserInputsIndex();
+        Optional<String> next = Optional.ofNullable(input());
+        return next;
+    }
+
+    private String input() {
+        String output = null;
+        if (userInputsIndex > -1 && userInputsIndex < userInputs.size()) {
+            output = userInputs.get(userInputsIndex);
+        }
+        return output;
+    }
+
+    public TextField getCommandTextField() {
+        return commandTextField;
+    }
+
+    public String getTextInCommandTextField() {
+        return commandTextField.getText();
+    }
+
     /**
```

Comment:
There seems to be a stray Javadoc comment here.

Reply:
There's a method underneath haha

Reply:
Oh XD